File: backend/app/schemas/affordability.py
from pydantic import BaseModel
from typing import Optional, List, Any, Dict


class AffordabilityCheckRequest(BaseModel):
    itemName: str
    itemPrice: float
    category: str
    isRecurring: bool


class AffordabilityResultOut(BaseModel):
    recommendation: str
    itemPrice: float
    availableBalance: float
    expectedMonthlyExpenses: float
    upcomingObligations: float
    savingsGoalContribution: float
    estimatedDisposableAmount: float
    aiExplanation: str


class AffordabilityRecordOut(BaseModel):
    id: str
    itemName: str
    itemPrice: float
    category: str
    isRecurring: bool
    date: str
    recommendation: str
    result: AffordabilityResultOut

    model_config = {"from_attributes": True}

File: src/types/affordability.ts
// ── Affordability types ──────────────────────────────────────────────────────
export type AffordabilityRecommendation =
  | 'comfortable'
  | 'proceed_with_caution'
  | 'not_recommended';

export interface AffordabilityCheckRequest {
  itemName: string;
  itemPrice: number;
  category: string;
  isRecurring: boolean;
}

export interface AffordabilityResult {
  recommendation: AffordabilityRecommendation;
  itemPrice: number;
  availableBalance: number;
  expectedMonthlyExpenses: number;
  upcomingObligations: number;
  savingsGoalContribution: number;
  estimatedDisposableAmount: number;
  aiExplanation: string;
}

export interface AffordabilityRecord {
  id: string;
  itemName: string;
  itemPrice: number;
  category: string;
  isRecurring: boolean;
  date: string;
  recommendation: AffordabilityRecommendation;
  result: AffordabilityResult;
}
